src/ranch.erl

```diff
@@ -315,51 +326,77 @@ set_protocol_options(Ref, Opts) ->
 
 -spec info() -> [{any(), [{atom(), any()}]}].
 info() ->
-	[{Ref, listener_info(Ref, Pid)}
-		|| {Ref, Pid} <- ranch_server:get_listener_sups()].
+	[{Ref, info(Ref)}
+		|| {Ref, _} <- ranch_server:get_listener_sups()].
 
 -spec info(ref()) -> [{atom(), any()}].
 info(Ref) ->
-	Pid = ranch_server:get_listener_sup(Ref),
-	listener_info(Ref, Pid).
-
-listener_info(Ref, Pid) ->
-	[_, Transport, _, Protocol, _] = ranch_server:get_listener_start_args(Ref),
-	ConnsSup = ranch_server:get_connections_sup(Ref),
-	Status = get_status(Ref),
-	{IP, Port} = get_addr(Ref),
-	MaxConns = get_max_connections(Ref),
-	TransOpts = ranch_server:get_transport_options(Ref),
-	ProtoOpts = get_protocol_options(Ref),
-	[
-		{pid, Pid},
-		{status, Status},
-		{ip, IP},
-		{port, Port},
-		{max_connections, MaxConns},
-		{active_connections, ranch_conns_sup:active_connections(ConnsSup)},
-		{all_connections, proplists:get_value(active, supervisor:count_children(ConnsSup))},
-		{transport, Transport},
-		{transport_options, TransOpts},
-		{protocol, Protocol},
-		{protocol_options, ProtoOpts}
-	].
+	[{Key, info(Ref, Key)}
+		|| Key <- [pid, status, ip, port, max_connections,
+			active_connections, all_connections, transport,
+			transport_options, protocol, protocol_options]].
+
+-spec info(ref(), pid) -> pid;
+	(ref(), status) -> running | suspended;
+	(ref(), ip) -> inet:ip_address();
+	(ref(), port) -> inet:port_number();
+	(ref(), max_connections) -> max_conns();
+	(ref(), active_connections) -> non_neg_integer();
+	(ref(), all_connections) -> non_neg_integer();
+	(ref(), transport) -> module();
+	(ref(), transport_options) -> opts();
+	(ref(), protocol) -> module();
+	(ref(), protocol_options) -> any().
+info(Ref, pid) ->
+	ranch_server:get_listener_sup(Ref);
+info(Ref, status) ->
+	get_status(Ref);
+info(Ref, ip) ->
+	{IP, _} = get_addr(Ref),
+	IP;
+info(Ref, port) ->
+	{_, Port} = get_addr(Ref),
+	Port;
+info(Ref, max_connections) ->
+	get_max_connections(Ref);
+info(Ref, active_connections) ->
+	get_connections(Ref, active);
+info(Ref, all_connections) ->
+	get_connections(Ref, all);
+info(Ref, transport) ->
+	[_, Transport, _, _, _] = ranch_server:get_listener_start_args(Ref),
+	Transport;
+info(Ref, transport_options) ->
+	ranch_server:get_transport_options(Ref);
+info(Ref, protocol) ->
+	[_, _, _, Protocol, _] = ranch_server:get_listener_start_args(Ref),
+	Protocol;
+info(Ref, protocol_options) ->
+	get_protocol_options(Ref).
```

Comment:
This change seems completely unrelated and not necessary (I didn't see any change other than active/all_connections retrieval method having changed). Correct? So probably better to keep it all as it was and just replace the two relevant calls.

Reply:
Huh, now how did _that_ end up in there... You're right, it is unrelated.

I had been playing with this quite a long time ago, then got distracted, then forgot. IIRC, my reason for this was so one could query a very specific info for a listener instead not all of it, which takes some extra time to collect only to be ignored,

Reply:
Feel free to update the PR to master and remove this, I'll get to it on Wednesday myself.